include/asterisk/channel.h

```diff
@@ -4076,6 +4077,7 @@ const char *ast_channel_linkedid(const struct ast_channel *chan);
 const char *ast_channel_parkinglot(const struct ast_channel *chan);
 const char *ast_channel_hangupsource(const struct ast_channel *chan);
 const char *ast_channel_dialcontext(const struct ast_channel *chan);
+const char *ast_channel_dial_announcement(const struct ast_channel *chan);
```

Comment:
I am not a fan of using the channel API or adding a stringfield for the sole purpose of communicating to app_dial that an announcement  is to be played on an interface when dial. I don't have an alternative of how to do it, but an alternative should be explored.

Reply:
It was the cleanest way I could think to do it at the time, but I agree it's slightly awkward.

I have two thoughts about this:
- A variable is already set on the channel containing the filename to play (since this is the most flexible, least core intrusive way to do it...). The API is simply to indicate that the file should be played. Is there an existing control frame that might be appropriate to use here? Alternately, would adding a new control frame altogether be more appropriate than the current implementation?
- If the answer to both of the above is 'no', would setting another variable on the channel to indicate that a file should be played be acceptable to you? I didn't like this approach because I suspect it will perform worse, given that every time the channel is polled, we'll have to traverse all the channel variables to check if this variable exists, making this linear time rather than a constant time operation. I would prefer an approach that doesn't have a performance impact.
- Finally, if the first bullet is not acceptable but the second is, how about using a control frame to signal to app_dial that "something interesting" has happened, and only then checking if the variable exists? I think there are things like SRCCHANGE that seemed to be used for things like this.

Reply:
There is no existing control frame specifically for this. There are other playback stream control frames, provided that it is done in a generic fashion and documented a control frame could be added alongside these to start playback.